Maple2.Model/Metadata/Table/ScrollTable.cs

```diff
@@ -16,7 +16,14 @@ public record EnchantScrollMetadata(
 
 public record ItemExchangeScrollTable(IReadOnlyDictionary<int, ItemExchangeScrollMetadata> Entries) : Table;
 
-public record ItemExchangeScrollMetadata();
+public record ItemExchangeScrollMetadata(
+    Ingredient RecipeScroll,
+    Ingredient RewardItem,
+    int TradeCountDeduction,
+    long RequiredMeso,
+    IReadOnlyList<Ingredient> RequiredItems);
+
+public record Ingredient(int ItemId, short Rarity, int Amount, ItemTag Tag);
```

Comment:
This is too common of a class name to have in the top-level namespace. If you want to keep this you should move it somewhere common and use it everywhere. (There are definitely other places where we need this struct) Otherwise just nest it in `ItemExchangeScrollMetadata`.

Also Rarity is commonly used as an `int`, but if you mostly need it as a `short` here that's fine.